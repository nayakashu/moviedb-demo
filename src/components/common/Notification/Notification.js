import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Notification.scss';

export class Notification extends Component {
  render() {
    const notifClasses = classNames('notif-container', this.props.type);

    return (
      <div className={notifClasses}>
        <div className="notif-message">
          <i className="material-icons icon-notif">notifications</i>Bomb lagdi
          menu
        </div>
        <i className="material-icons btn-close">close</i>
      </div>
    );
  }
}

Notification.propTypes = {
  type: PropTypes.string
};

export default Notification;
