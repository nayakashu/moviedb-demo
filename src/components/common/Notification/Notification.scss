@import '../../../Common.scss';

.notif-container {
  border: none;
  width: 550px;
  position: fixed;
  top: 114px;
  right: 100px;
  color: $white;
  font-size: 15px;
  padding: 15px 20px 20px 20px;
  box-shadow: 2px 2px 5px $shadow-color;

  i.icon-notif {
    font-size: 20px;
    margin-right: 10px;
    position: relative;
    top: 5px;
  }

  i.btn-close {
    float: right;
    position: absolute;
    top: 17px;
    right: 17px;
  }
}

.success {
  background: $success-modal-bg;
}

.danger {
  background: $danger-modal-bg;
}
