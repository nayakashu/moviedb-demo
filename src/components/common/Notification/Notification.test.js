import React from 'react';
import { shallow } from 'enzyme';
import { Notification } from './Notification';

describe('Notification', () => {
  const notification = shallow(<Notification />);

  it('renders correctly', () => {
    expect(notification).toMatchSnapshot();
  });
});
