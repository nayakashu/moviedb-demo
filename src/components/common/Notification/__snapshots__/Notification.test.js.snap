// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notification />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="notif-message"
        >
          <i
            className="material-icons icon-notif"
          >
            notifications
          </i>
          Bomb lagdi menu
        </div>,
        <i
          className="material-icons btn-close"
        >
          close
        </i>,
      ],
      "className": "notif-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="material-icons icon-notif"
            >
              notifications
            </i>,
            "Bomb lagdi menu",
          ],
          "className": "notif-message",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "notifications",
              "className": "material-icons icon-notif",
            },
            "ref": null,
            "rendered": "notifications",
            "type": "i",
          },
          "Bomb lagdi menu",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "material-icons btn-close",
        },
        "ref": null,
        "rendered": "close",
        "type": "i",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="notif-message"
          >
            <i
              className="material-icons icon-notif"
            >
              notifications
            </i>
            Bomb lagdi menu
          </div>,
          <i
            className="material-icons btn-close"
          >
            close
          </i>,
        ],
        "className": "notif-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="material-icons icon-notif"
              >
                notifications
              </i>,
              "Bomb lagdi menu",
            ],
            "className": "notif-message",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "notifications",
                "className": "material-icons icon-notif",
              },
              "ref": null,
              "rendered": "notifications",
              "type": "i",
            },
            "Bomb lagdi menu",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "material-icons btn-close",
          },
          "ref": null,
          "rendered": "close",
          "type": "i",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
